components/log4j/org.wso2.carbon.analytics.shared.data.agents.log4j/src/main/java/org/wso2/carbon/analytics/shared/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -215,82 +220,38 @@ protected void append(LoggingEvent event) {
         if (LogAppenderServiceValueHolder.getConfigurationContextService() != null) {
             if (isFirstEvent) {
                 publisherInitializer();
-                if (dataPublisher == null) {
-                    LogLog.error("Data Publisher not initialize for url : " + url);
-                }
                 isFirstEvent = false;
             }
 
             if (dataPublisher != null) {
                 Logger logger = Logger.getLogger(event.getLoggerName());
-                TenantDomainAwareLoggingEvent tenantEvent;
-                if (event.getThrowableInformation() != null) {
-                    tenantEvent = new TenantDomainAwareLoggingEvent(event.fqnOfCategoryClass, logger, event.timeStamp,
-                            event.getLevel(), event.getMessage(), event.getThrowableInformation().getThrowable());
-                } else {
-                    tenantEvent = new TenantDomainAwareLoggingEvent(event.fqnOfCategoryClass, logger, event.timeStamp,
-                            event.getLevel(), event.getMessage(), null);
-                }
-                tenantId = AccessController.doPrivileged(new PrivilegedAction<Integer>() {
-                    public Integer run() {
-                        return CarbonContext.getThreadLocalCarbonContext().getTenantId();
-                    }
-                });
+                ThrowableInformation throwableInformation = event.getThrowableInformation();
+                TenantDomainAwareLoggingEvent tenantEvent = new TenantDomainAwareLoggingEvent(event.fqnOfCategoryClass,
+                        logger, event.timeStamp, event.getLevel(), event.getMessage(),
+                        (throwableInformation != null) ? throwableInformation.getThrowable() : null);
 
-                tenantDomain = AccessController.doPrivileged(new PrivilegedAction<String>() {
-                    public String run() {
-                        return CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-
-                    }
-                });
-
-                tenantEvent.setTenantDomain(tenantDomain);
-
-                if (tenantId == MultitenantConstants.INVALID_TENANT_ID) {
-                    if (tenantDomain != null && !tenantDomain.equals("")) {
-                        try {
-                            tenantId = getTenantIdForDomain(tenantDomain);
-                        } catch (UserStoreException e) {
-                            LogLog.error("Cannot find tenant id for the given tenant domain.", e);
-                        }
-                    }
-                }
-                appName = CarbonContext.getThreadLocalCarbonContext().getApplicationName();
+                int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true);
                 tenantEvent.setTenantId(String.valueOf(tenantId));
+                String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain(true);
```

Comment:
The PrivilegedCarbonContext is used to set the value and to retrieve the CarbonContext is used.